src/Avalonia.X11/X11Window.cs

```diff
@@ -565,7 +569,7 @@ ev.ButtonEvent.button switch
                 {
                     if (ev.ClientMessageEvent.ptr1 == _x11.Atoms.WM_DELETE_WINDOW)
                     {
-                        if (Closing?.Invoke(WindowCloseReason.WindowClosing) != true)
+                        if (IsEnabled && Closing?.Invoke(WindowCloseReason.WindowClosing) != true)
```

Comment:
I can't really review linux specific PRs with confidence, but is it safe to disable window closing for Windows with IsEnabled=false? It will affect more than just dialog owners. Should be child window closed with the owner together?
cc @kekekeks @jmacato

Reply:
This code only affects events from the OS window manager and prevents users from closing disabled windows (e.g. owner of the modal window). You can still close a disabled window via API (Window.Close).